provider/app.py

```diff
@@ -31,27 +31,36 @@
 def postTrigger(namespace, trigger):
     body = request.get_json(force=True, silent=True)
     triggerFQN = '/' + namespace + '/' + trigger
-    logging.info("[{}] Ensuring user has access rights to post a trigger".format(triggerFQN))
-
-    trigger_get_response = requests.get(body["triggerURL"])
-    trigger_get_status_code = trigger_get_response.status_code
-    logging.info("[{}] Repsonse status code from trigger authorization {}".format(triggerFQN, trigger_get_status_code))
-
-    if trigger_get_status_code == 200:
-        logging.info("[{}] User authenticated. About to create consumer {}".format(triggerFQN, str(body)))
-        createAndRunConsumer(triggerFQN, body)
-        logging.info("[{}] Finished creating consumer.".format(triggerFQN))
-        result = {'success': True}
-    elif trigger_get_status_code == 401:
-        logging.info("[{}] User not authorized to post trigger".format(triggerFQN))
+
+    if triggerFQN in consumers:
+        logging.info("[{}] Trigger already exists".format(triggerFQN))
+        trigger_get_status_code = 403
```

Comment:
@dubeejw I suppose there is some room for interpretation here, but I prefer to interpret this as a bad request on the client because they asked to do something that does not need to be done. To me, this is an invalid request, rather than one they are not allowed to perform. Maybe another way to say it is that this exact request would likely be allowed if the trigger didn't already exist, but 403 says "you can't do this, don't ever try again."